packages/components/src/list/experimental-list-item.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { CSSTransition } from 'react-transition-group';
+
+/**
+ * Internal dependencies
+ */
+import { handleKeyDown } from './list-item';
+
+type CSSTransitionProps = {
+	in: boolean;
+	exit: boolean;
+	enter: boolean;
+	onExited: () => void;
+};
+
+type ListItemProps = {
+	onClick?: () => void;
```

Comment:
On "click" seems misleading considering this is attached to `KeyboardEvent`.

Reply:
I think you've just spotted a mistake in the code. it should be attached to both. Attaching it to keyboard is to satisfy keyboard nav requirements but it should be attached as the onClick handler as well. thanks for spotting this, I should be able to cover this with a test too.

Reply:
I ended up removing this from being declared directly in the props since now I'm defaulting to acting like a link. now onClick is attached as part of spreading `otherProps`